File: 3 CSV IO/python/csvIO.py
import pandas as pd
import numpy as np
import time

N = 1
read_time = []
write_time = []

for i in range(N):
    tic = time.time()
    M = pd.DataFrame.from_csv(f"C:/pseudorandomcoder/data_set/dataset{i}.csv")
    read_time.append(time.time() - tic)
    print(f"read dataset{i}.csv")
    
    tic = time.time()
    M.to_csv(f"C:/pseudorandomcoder/data_set/python/dataset{i}_out.csv")
    write_time.append(time.time() - tic)
    print(f"wrote dataset{i}_out.csv")

print(f"Average Read Time: {np.mean(read_time)}")
print(f"Average Write Time: {np.mean(write_time)}")

File: 3 CSV IO/matlab/csvIO.m
clc; clear; close all;
N = 1;
read_time = zeros(N, 1);
write_time = zeros(N, 1);

for i = 1:N
    tic
    M = csvread("C:/pseudorandomcoder/data_set/dataset" + num2str(i) + ".csv");
    read_time(i) = toc;
    fprintf("read dataset%d.csv\n", i);
    
    tic
    % csvwrite("C:/pseudorandomcoder/data_set/matlab/dataset_out" + num2str(i) + ".csv", M);
    dlmwrite("C:/pseudorandomcoder/data_set/matlab/dataset_out" + num2str(i) + ".csv", M, 'precision', 16);
    write_time(i) = toc;
    fprintf("wrote dataset%d_out.csv\n", i);
    break
end

fprintf("Average Read Time: %f\n", mean(read_time))
fprintf("Average Write Time: %f\n", mean(write_time))